src/components/Header.jsx

```diff
@@ -1,108 +1,133 @@
-export default function Header({ user, theme, setTheme }) {
-    const handleCheckChange = () => {
-      if(theme === 'dark') {
-        setTheme('light');
-      } else {
-        setTheme('dark');
-      }
-    }
+import React, { useContext } from "react";
+// 6. Import AppContext
+import { AppContext, ThemeContext } from "../App";
+
+// 4. Remove props
+export default function Header() {
+  // 5. Add useContext here
+
+  const { user } = useContext(AppContext);
+  const { theme, setTheme } = useContext(ThemeContext);
 
-    const handleButtonClick = () => {
-      console.log("CLICK!");
+  const handleCheckChange = () => {
+    let t = theme;
+    if (theme === "dark") {
+      setTheme("light");
+      t = "light";
+    } else {
+      setTheme("dark");
+      t = "dark";
     }
+    localStorage.setItem("TwitterTheme", t);
```

Comment:
I like that how you've set a variable for theme, and you're using it to set the `localStorage`. You could also do this with your theme state. Like so:
```js
  if (theme === "dark") {
      t = "light";
    } else {
      t = "dark";
    }
    setTheme(t);
    localStorage.setItem("TwitterTheme", t);
```